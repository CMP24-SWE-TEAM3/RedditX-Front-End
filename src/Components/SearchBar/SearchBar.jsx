// imports
import React from 'react';
import {StyledSearchButton, StyledSearchIcon} from "./SearchBar.styled";
import InputForm from "Components/InputForm/InputForm";

/**
 * Component that displays the search results for a given search term.
 *
 * @returns {React.Component}
 */
const SearchBar = (props) => {
    return (
        <>
            <StyledSearchButton>
                <StyledSearchIcon/>
                <InputForm/>
            </StyledSearchButton>
        </>
    );
};

export default SearchBar;
