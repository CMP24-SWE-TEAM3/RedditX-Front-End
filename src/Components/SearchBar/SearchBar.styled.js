import styled from 'styled-components'
import {CiSearch} from "react-icons/ci";

export const StyledSearchButton = styled.div`
  display: flex;
  border-radius: 20px;
  height: 40px;
  //width: 35%;
  background: ${({theme}) => theme.background.post_background};
  margin: auto;
  //width:100%;
  @media (min-width: 1286px) {
    min-width: 516px;
  }

`
export const StyledSearchIcon = styled(CiSearch)`
  color: ${({theme}) => theme.color.muted};
  position: absolute;
  margin-top: 10px;
  min-width: 50px;
`