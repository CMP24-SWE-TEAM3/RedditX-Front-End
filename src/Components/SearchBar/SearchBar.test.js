import {shallow} from 'enzyme';

// import Components
import SearchBar from "./SearchBar";

describe("Search Bar Component", () => {
    it("this is a test for Search", () => {
        expect(shallow(<SearchBar/>)).toMatchSnapshot();

    });
});