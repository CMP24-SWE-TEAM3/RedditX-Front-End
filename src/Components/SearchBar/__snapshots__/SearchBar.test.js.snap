// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Component this is a test for Search 1`] = `ShallowWrapper {}`;
